packages/components/src/components/drawer/drawer.jsx

```diff
@@ -15,54 +15,45 @@ const IconDrawer = ({ className }) => (
     </svg>
 );
 
-class Drawer extends React.PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = { is_open: props.is_open };
-    }
+const Drawer = ({
+    anchor = 'left',
+    children,
+    className,
+    is_open,
+    toggleDrawer,
+    contentClassName,
+    footer,
+    header,
+    width = 250,
+    zIndex = 4,
+}) => {
+    const [will_open, set_is_open] = React.useState(is_open);
```

Comment:
```suggestion
    const [will_open, set_is_open] = React.useState(is_open);
```

You can use the `useStateCallback` hook to replace this 😄  reasons are as below

Reply:
done